xdem/biascorr.py

```diff
@@ -0,0 +1,113 @@
+"""Bias corrections for DEMs"""
+from __future__ import annotations
+
+from typing import Callable, Optional
+
+import numpy as np
+import rasterio as rio
+
+import xdem
+
+
+class DirectionalBias(xdem.coreg.Coreg):
+    """
+    For example for DEM along- or across-track bias correction.
+    """
+
+    def __init__(self, bias_func : Callable = xdem.robust_stats.robust_polynomial_fit):  # pylint: disable=super-init-not-called
+        """
+        Instantiate an directional bias correction object.
+
+        :param bias_func: The function to fit the bias. Default: robust polynomial of degree 1 to 6.
+        """
+        super().__init__(meta={"bias_func": bias_func})
+
+    def _fit_func(self, ref_dem: np.ndarray, tba_dem: np.ndarray, transform: Optional[rio.transform.Affine],
+                  weights: Optional[np.ndarray], angle: Optional[float] = None, verbose: bool = False,**kwargs):
+        """Estimate the bias using the bias_func."""
+
+        if verbose:
+            print('Getting directional coordinates')
+
+        diff = ref_dem - tba_dem
+        x, _ = xdem.spatial_tools.get_rotated_xy(ref_dem,angle=angle)
+
+        if verbose:
+            print("Estimating directional bias correction with function "+ self.meta['bias_func'].__name__)
+        deg, coefs = self.meta["bias_func"](x,diff,**kwargs)
+
+        if verbose:
+            print("Directional bias estimated")
+
+        self._meta['angle'] = angle
+        self._meta['degree'] = deg
+        self._meta["coefs"] = coefs
+
+    def _to_matrix_func(self) -> np.ndarray:
+        """Convert the bias to a transform matrix."""
+
+        raise ValueError(
```

Comment:
This is necessary to not raise a `NotImplementedError`. However, it would probably be better if there was a generic way to do this, no?

If you added this to line 747 in main:
```python
if self._is_affine == False:
     raise ValueError(f"Non-affine coreg class ({type(self)}) cannot be represented by transformation matrices.")
```

then this would work better for all non-affine coreg subclasses. The only thing one has to specify in the `__init__` is `self._is_affine = False`.

Reply:
Zgreed. All of this is added!